maa-cli/src/run/copilot.rs

```diff
@@ -0,0 +1,207 @@
+
+    // Visit the API to obtain the data of homework.
+    let client = Client::new();
+    let response = client.get(&url).send().unwrap();
+    let json: JsonValue = response.json().unwrap();
+    let status_code = json["status_code"].clone().to_string();
+    if status_code != "200" {
+        println!("Request failed");
+        return false;
+    }
+
+    // Extract the response
+    let context = json["data"]["content"].clone();
+    let content = match context.as_str() {
+        Some(content) => content.to_string(),
+        None => {
+            println!("Error parsing JSON");
+            return false;
+        }
+    };
+
+    let recast: JsonValue = serde_json::from_str(&content).unwrap();
+
+    // Determine type of stage
+    let mut task_type: &str = "Copilot";
+    let content_type = recast["type"].clone();
+    if content_type == "SSS" {
+        task_type = "SSSCopilot";
+    }
+
+    // Print stage info
+    println!("Stage info:\n");
+    if task_type == "Copilot" {
+        let stage_code_name = recast["stage_name"].as_str().unwrap();
+        let stage_name = find_stage_name(stage_code_name, stage_dir);
+        println!("{}\n", stage_name);
+    } else {
+        let stage_name = recast["stage_name"].as_str();
+        println!("{}\n", stage_name.unwrap());
+    }
+
+    //Print operators info
+    println!("Operator lists:\n");
+    let opers = recast["opers"].clone();
+    for operator in opers.as_array().unwrap() {
+        let name = operator["name"].as_str().unwrap();
+        let skill = operator["skill"].as_u64().unwrap();
+        println!("{}, Skill: {};", name, skill,);
+    }
+
+    //Get input of user
+    let mut formation: bool = true;
+    let mut loop_times: i64 = 1;
+    if task_type == "Copilot" {
+        let input: Input<String> = Input::new(
+            Some("yes".to_string()),
+            Some("Whether to quick build(Y/N):"),
+        );
+        let stdin = io::stdin();
+        let stdout = io::stdout();
+        let _ = input.prompt(stdout.lock());
+        let result = input.ask(stdout.lock(), stdin.lock());
+        match result.unwrap().as_str() {
+            "yes" | "Yes" | "y" | "Y" => {
+                formation = true;
+            }
+            "no" | "No" | "n" | "N" => {
+                formation = false;
+            }
+            _ => {
+                println!("Invaild input");
+                return false;
+            }
+        }
+    } else {
+        let input: Input<i64> = Input::new(Some(1), Some("Loop times:"));
+        let stdin = io::stdin();
+        let stdout = io::stdout();
+        let _ = input.prompt(stdout.lock());
+        let result = input.ask(stdout.lock(), stdin.lock());
+        match result {
+            Ok(value) => {
+                loop_times = value;
+            }
+            Err(_error) => {
+                println!("Invaild input");
+                return false;
+            }
+        }
+    }
```

Comment:
I believe we shouldn't prompt the user for input manually here; the `Input` should be handled in the `run::run`. The only issue is that `Input<bool>` might not be ideal. I'm considering implementing a separate input type specifically for `bool`.